menas/src/main/scala/za/co/absa/enceladus/menas/controllers/DatasetController.scala

```diff
@@ -53,4 +60,97 @@ class DatasetController @Autowired() (datasetService: DatasetService)
+                                ): CompletableFuture[Map[String, String]] = {
+    getPropertiesForVersionCommon(datasetName, datasetVersion, allFilterParams)
+  }
+
+  @GetMapping(Array("/{datasetName}/properties"))
+  @ResponseStatus(HttpStatus.OK)
+  def getAllProperties(@PathVariable datasetName: String,
+                       @RequestParam allFilterParams: util.Map[String, String]): CompletableFuture[Map[String, String]] = {
+    datasetService.getLatestVersionNumber(datasetName).flatMap { version =>
+      getPropertiesForVersionCommon(datasetName, version, allFilterParams)
+    }
+  }
+
+  private def getPropertiesForVersionCommon(datasetName: String,
+                                            datasetVersion: Int,
+                                            allFilterParams: util.Map[String, String]): Future[Map[String, String]] = {
+    datasetService.getVersion(datasetName, datasetVersion).flatMap {
+      case Some(entity) => {
+        val dsProperties = entity.propertiesAsMap
+
+        val scalaFilterMap: Map[String, String] = allFilterParams.toScalaMap // in Scala 2.12, we could just do .asScala.toMap
+        if (scalaFilterMap.isEmpty) {
+          Future.successful(dsProperties) // no filtering, may even contain properties without existing definition
+        } else {
+          datasetService.filterProperties(dsProperties, DatasetController.paramsToPropertyDefinitionFilter(scalaFilterMap))
+        }
+      }
+      case None => throw notFound()
+    }
+  }
+
+  @PutMapping(Array("/{datasetName}/properties"))
+  @ResponseStatus(HttpStatus.CREATED)
+  def replaceProperties(@AuthenticationPrincipal principal: UserDetails,
+                        @PathVariable datasetName: String,
+                        @RequestBody newProperties: Map[String, String]): CompletableFuture[ResponseEntity[Option[Dataset]]] = {
+
+    datasetService.replaceProperties(principal.getUsername, datasetName, newProperties).map { dataset =>
+      dataset match {
+        case None => throw notFound()
+        case Some(ds) => {
+          val location: URI = new URI(s"/api/dataset/${ds.name}/${ds.version}")
+          ResponseEntity.created(location).body(dataset)
+        }
+      }
+    }
+  }
+
+  @GetMapping(Array("/{datasetName}/{datasetVersion}/properties/valid"))
+  @ResponseStatus(HttpStatus.OK)
+  def getPropertiesValidation(@PathVariable datasetName: String, @PathVariable datasetVersion: Int): CompletableFuture[Validation] = {
+    datasetService.getVersion(datasetName, datasetVersion).flatMap {
+      case Some(entity) => datasetService.validateProperties(entity.propertiesAsMap)
+      case None => throw notFound()
+    }
+  }
+
+}
+
+object DatasetController {
+  protected val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  private[controllers] def paramsToPropertyDefinitionFilter(filterParams: Map[String, String]): PropertyDefinition => Boolean = {
```

Comment:
This may be an overkill provided we now support just one filter (`put_into_info_file`), but should support easy extensions. Should I simplify or keep the support?